casanovo/denovo/dataloaders.py

```
@@ -24,6 +24,34 @@
+    -------
+    str
+        The scan number as a string, or ``""`` if absent.
+    """
+    try:
+        params = spectrum_dict.get("params", {})
+        for key in ("scans", "scan", "scan id"):
+            val = params.get(key)
+            if val is not None and str(val).strip():
+                return str(val).strip()
+    except Exception:
+        pass
```

Ruff (0.15.4)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)